File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/chat/smiley/LocalizedChatSmileyMessage.scala





















// Generated on 05/03/2013 10:34:38
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait LocalizedChatSmileyMessage extends ChatSmileyMessage {
  override val messageId = 6185

  protected trait Message extends ChatSmileyMessage.Message with Serializable {
    override val messageId = LocalizedChatSmileyMessage.messageId

    val cellId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(cellId)
    }

  }

  protected def create(entityId: Int, smileyId: Byte, accountId: Int, cellId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val entityId = buf.readInt()
    val smileyId = buf.readByte()
    val accountId = buf.readInt()
    val cellId = buf.readShort()

    create(entityId, smileyId, accountId, cellId)
  }

}

object LocalizedChatSmileyMessage extends LocalizedChatSmileyMessage {
  case class MessageImpl(entityId: Int, smileyId: Byte, accountId: Int, cellId: Short) extends Message
  type Target = MessageImpl

  override def create(entityId: Int, smileyId: Byte, accountId: Int, cellId: Short) = MessageImpl(entityId, smileyId, accountId, cellId)

  def apply(entityId: Int, smileyId: Byte, accountId: Int, cellId: Short) = create(entityId, smileyId, accountId, cellId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/LocalizedChatSmileyMessage.java


// Generated on 05/08/2013 19:37:43
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class LocalizedChatSmileyMessage extends ChatSmileyMessage {
    public static final int MESSAGE_ID = 6185;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short cellId;
    
    public LocalizedChatSmileyMessage() { }
    
    public LocalizedChatSmileyMessage(int entityId, byte smileyId, int accountId, short cellId) {
        super(entityId, smileyId, accountId);
        this.cellId = cellId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(cellId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        cellId = buf.readShort();
        if (cellId < 0 || cellId > 559)
            throw new RuntimeException("Forbidden value on cellId = " + cellId + ", it doesn't respect the following condition : cellId < 0 || cellId > 559");
    }
    
}
